packages/contractkit/src/identity/offchain/storage-writers.ts

```diff
@@ -1,20 +1,20 @@
 import { execSync } from 'child_process'
 import { mkdirSync, writeFile } from 'fs'
-import { parse } from 'path'
+import { normalize, parse } from 'path'
 
 export abstract class StorageWriter {
-  abstract write(_data: string, _dataPath: string): Promise<void>
+  abstract write(_data: Buffer, _dataPath: string): Promise<void>
 }
 
 export class LocalStorageWriter extends StorageWriter {
   constructor(readonly root: string) {
     super()
   }
-  async write(data: string, dataPath: string): Promise<void> {
+  async write(data: Buffer, dataPath: string): Promise<void> {
     return this.writeToFs(data, dataPath)
   }
 
-  protected async writeToFs(data: string, dataPath: string): Promise<void> {
+  protected async writeToFs(data: string | Buffer, dataPath: string): Promise<void> {
     await new Promise((resolve, reject) => {
       const directory = parse(dataPath).dir
       mkdirSync(this.root + directory, { recursive: true })
```

Comment:
If I understand JS Promises, we should use the callback version here to avoid blocking. Additionally, I believe the sync versions here may throw, which is nasty because `reject` will never get called.